raft/src/test/java/org/apache/kafka/raft/MockLog.java

```diff
@@ -41,7 +41,7 @@
 public class MockLog implements ReplicatedLog {
     private final List<EpochStartOffset> epochStartOffsets = new ArrayList<>();
     private final List<LogBatch> log = new ArrayList<>();
-    private long highWatermark = 0L;
+    private LogOffsetMetadata highWatermark = new LogOffsetMetadata(0L);
```

Comment:
I'm thinking a little bit about how we could validate offset metadata with the MockLog implementation. One idea is to give each `LogEntry` a unique ID, which could then be returned as a MockOffsetMetadata. Then we would be able to validate whether the information is getting retained correctly.